pypeit/core/trace.py

```diff
@@ -0,0 +1,1396 @@
+        if maxshift is not None:
+            xmsk[large_shift] = bitmask.turn_on(xmsk[large_shift], 'LARGESHIFT')
+
+    # Return the new centers, errors, and flags
+    return xfit, xerr, indx if bitmask is None else xmsk
+
+
+def fit_trace(flux, trace_cen, order, ivar=None, bpm=None, trace_bpm=None, weighting='uniform',
+              fwhm=3.0, maxshift=None, maxerror=None, function='legendre', maxdev=5.0, maxiter=25,
+              niter=9, bitmask=None, debug=False, idx=None, xmin=None, xmax=None):
+    """
+    Iteratively fit the trace of a feature in the provided image.
+
+    Each iteration performs two steps:
+        - Remeasure the trace centroid using :func:`masked_centroid`.
+          The size of the integration window (see the definition of
+          the `width` parameter for
+          :func:`pypeit.core.moment.moment1d`) depends on the type of
+          weighting: For *uniform weighting*, the code does a third
+          of the iterations with `width = 2*1.3*fwhm`, a third with
+          `width = 2*1.1*fhwm`, and a third with `width = 2*fwhm`.
+          For *Gaussian weighting*, all iterations use `width =
+          fwhm/2.3548`.
+        - Fit the centroid measurements with a 1D function of the
+          provided order. See :func:`pypeit.core.pydl.TraceSet`.
+
+    The number of iterations performed is set by the keyword argument
+    `niter`. There is no convergence test, meaning that this number
+    of iterations is *always* performed.
+
+    History:
+        23-June-2018  Written by J. Hennawi
+
+    Args:
+        flux (`numpy.ndarray`_):
+            Image to use for tracing. Must be 2D with shape (nspec,
+            nspat).
+        trace_cen (`numpy.ndarray`_):
+            Initial guesses for trace centroids in the spatial
+            dimension. This can either be an 2-d array with shape
+            (nspec, nTrace) array, or a 1-d array with shape (nspec)
+            for the case of a single trace.
+        order (:obj:`int`):
+            Order of function to fit to each trace.  See `function`.
+        ivar (`numpy.ndarray`_, optional):
+            Inverse variance of the image intensity.  If not provided,
+            unity variance is used.  If provided, must have the same
+            shape as `flux`.
+        bpm (`numpy.ndarray`_, optional):
+            Boolean array with the input bad-pixel mask for the image
+            (pixels to ignore are True). If not provided, all values
+            in `flux` are considered valid. If provided, must have
+            the same shape as `flux`.
+        trace_bpm (`numpy.ndarray`_, optional):
+            Boolean array with the trace bad-pixel mask; i.e., places
+            where you know the trace is going to be bad that you
+            always want to mask in the fits. Shape must match
+            `trace_cen`.
+        weighting (:obj:`str`, optional):
+            The weighting to apply to the position within each
```

Comment:
I kind of agree with this, but I want to push back a little for the following reason.  If the options change in `moment1d`, I'll have to remember to change the docs in multiple places.  The point of having the
```
`:func:`pypeit.core.moment.moment1d`
```
syntax in the docstring is that Sphinx uses that to create a link to the documentation for that function.  So if users want to see the options they can just click on the link.  However, the counter argument is that is an added burden on the user.  I'd rather not change this because I lean on users being able to click around the documentation for this kind of information in multiple places.

Reply:
Okay numpy adopts similar sparseness to docs, and I find it annoying to click around to find those options, because sometimes I cannot find them that easily. Anyway, the counter argument is that if the options change, you only have to change them in one place. I'm fine either way.